20231212-modular-models.md

```diff
@@ -0,0 +1,229 @@
+# Modular Models
+
+## Meta
+
+- **Name**: Enable defining models in multiple files
+- **Start Date**: 2023-12-12
+- **Author(s)**: [aaguiarz](https://github.com/aaguiarz)
+- **Status**: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- **PR Link**: https://github.com/openfga/rfcs/pull/14/files
+- **Relevant Issues**:
+- **Supersedes**: N/A
+
+## Summary
+
+The FGA model currently needs to be defined in a single file. This RFC proposes a way to define the model in multiple files that can be combined into a single model when writing it to a FGA store.
+
+## Definitions
+
+- [OpenFGA DSL](https://openfga.dev/docs/configuration-language)
+
+## Motivation
+
+### Enable multiple teams collaborate on the same conceptual model
+
+Authorization logic is application/service specific. In companies with multiple teams building multiple components, each team should own the authorization logic for their domain.
+
+Having all teams to maintain a model in a single file has several drawbacks:
+
+  - Requires a more comprehensive PR process where all teams need to be involved.
+  - The model can become large and difficult to understand.
+  - It's likely that the authorization model can't be part of the same repository as the code that uses it.
+
+### Enable different applications to perform different actions in different modules
+
+In addition to enable different developers maintain their own modules, it should be possible to enable different applications to perform different actions in different modules. 
+
+For example, the application that owns a module should be able to write/delete tuples from the store, but other applications could only be able to query it.
```

Comment:
For example, applications belonging to the domain which owns a module should be able to write/delete tuples of the object types the module defines, but other applications outside of that domain should only be able to query relationships of those object types.